File: php/p020.php
<?php

/**
 * Problem 20 - Factorial digit sum
 * 
 * Lul php can only store 32bit ints. Gonna do this one in python rather than
 * have to deal with keeping track of big ints manually.
 */

 function factorial($n) {
     if ($n === 1) {
         return 1;
     } else {
         return $n * factorial($n - 1);
     }
}

echo factorial(100);

File: python/p020.py
"""Problem 20 - Factorial digit sum

TODO: desc
TODO: put factoiral function in lib and handle invalid input
"""

def factorial(n):
    if n == 1:
        return 1
    else:
        return n * factorial(n-1)

print(sum([int(c) for c in str(factorial(100))]))
